src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -100,6 +100,84 @@ pub struct RSAggregateResult {
     pub type_mask: RSResultTypeMask,
 }
 
+impl RSAggregateResult {
+    /// The number of results in this aggregate result
+    pub fn len(&self) -> usize {
+        self.num_children as _
+    }
+
+    /// Check whether this aggregate result is empty
+    pub fn is_empty(&self) -> bool {
+        self.num_children == 0
+    }
+
+    /// The capacity of the aggregate result
+    pub fn capacity(&self) -> usize {
+        self.children_cap as _
+    }
+
+    /// The current type mask of the aggregate result
+    pub fn type_mask(&self) -> RSResultTypeMask {
+        self.type_mask
+    }
+
+    /// Get an iterator over the children of this aggregate result
+    pub fn iter(&self) -> RSAggregateResultIter {
+        RSAggregateResultIter {
+            agg: self,
+            index: 0,
+        }
+    }
+
+    /// Get the child at the given index, if it exists
+    ///
+    /// # Safety
+    /// The caller must ensure that the memory at the given index is still valid
+    pub fn get(&self, index: usize) -> Option<&RSIndexResult> {
+        if index < self.num_children as usize {
+            // SAFETY: We are guaranteed that the index is within bounds because of the check above.
+            let result_ptr = unsafe { self.children.add(index) };
+
+            // SAFETY: It is safe to dereference the pointer because we correctly got it using
+            // `add` above.
+            let result_addr = unsafe { *result_ptr };
+
+            // SAFETY: The caller is to guarantee that the memory at `result_addr` is still valid.
+            Some(unsafe { &*result_addr })
+        } else {
+            None
+        }
+    }
+
+    /// Reset the aggregate result, clearing all children and resetting the type mask
+    pub fn reset(&mut self) {
+        self.num_children = 0;
+        self.type_mask = RSResultTypeMask::empty();
+    }
+}
+
+pub struct RSAggregateResultIter<'a> {
```

Comment:
I'd document this if only to have full doc coverage.

Reply:
Thin but simple https://github.com/RediSearch/RediSearch/pull/6492/commits/985fbb4f5f28f2fd1e4a0f4566fd119d80e44add